server/src/admin/admin.data-access.ts

```diff
@@ -51,3 +109,47 @@ export const updateReviewVisibility = async (
 ) => {
   await Reviews.updateOne({ _id: reviewId }, { $set: { visible, reported } });
 };
+
+/*
+ * Functions for the manage admin functionality. Enables grant admin privilege to a
+ * user, removing admin privilege from a user, and retrieving all priviliged users.
+ * Admin users have privilege "admin" and everyone else has privilege "regular"
+ */
+
+export const findAdminUsers = async () => {
+  const adminUsers = await Students.find({ privilege: 'admin' }).exec()
+  return adminUsers
+}
+
+export const removeAdminPrivilege = async (id: string) => {
+  const res = await Students.updateOne({ _id: id }, { $set: {privilege: 'regular'} }).exec()
+  return res
+}
+
+export const grantAdminPrivilege = async (id: string) => {
+  const res = await Students.updateOne({ netId: id }, { $set: {privilege: "admin"} }).exec()
+  return res
+}
+
+/*
+ * If there is an attempt to grant admin privilege to someone not in the database,
+ * a new user will be created with the given netid and added to the database.
+ */
+
+export const createNewAdminUser = async (id: string) => {
+
+  const admin: InsertStudentType = {
+    _id: shortid.generate(),
+    firstName: '',
+    lastName: '',
+    netId: id,
+    affiliation: '',
+    token: '',
+    privilege: 'admin',
+  };
+  
+  const newAdmin = new Students(admin);
+  const res = await newAdmin.save();
+
+  return res
+}
```

Comment:
There should probably be some "security measure" here so that we don't accidentally add a random member as admin. I'm wondering if there's a way to cross-check with like a CUReviews roster automatically.